app/mender.go

```diff
@@ -466,6 +466,34 @@ func (m *Mender) TransitionState(to State, ctx *StateContext) (State, bool) {
 	return transitionState(to, ctx, m)
 }
 
+func spinEventLoop(c *ControlMapPool, ctx *StateContext, to State) (State, bool) {
+	for {
+		switch to.Transition() {
+		case ToArtifactReboot_Enter, ToArtifactCommit_Enter, ToArtifactInstall:
+			action := c.QueryAndUpdate(to.Transition().String())
+			switch action {
+			case "continue":
+				return to, false
+			case "pause":
+				// wait until further notice
+				<-c.Updates
+			case "fail":
+				// Fail (the update) immediately
+				// TODO/FIXME
+				// Simply set the to state to failed ? Hmmm... not sure if this is right
+				sd, err := datastore.LoadStateData(ctx.Store)
```

Comment:
Yeah, it feels like the wrong place to do this. But each state has a `HandleError` function which is called whenever a state script fails. In principle this is the same kind of failure, so could we reuse that? See [this part](https://github.com/mendersoftware/mender/pull/768/commits/a578cefe4abe503915908405d5562c9eb3702f0f#diff-31fdb7c71693c702fb30629bde922bafc9da308305ac0330597849769d7e1dedR533-R537).